easybuild/easyblocks/generic/gopackage.py

```diff
@@ -0,0 +1,169 @@
+        })
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables."""
+        super(GoPackage, self).__init__(*args, **kwargs)
+        self.go_cmd = None
+
+    def prepare_go(self):
+        """Go-specific preperations."""
+
+        go = None
+        go_root = get_software_root('Go')
+        if go_root:
+            bin_go = os.path.join(go_root, 'bin', 'go')
+            if os.path.exists(bin_go) and os.path.samefile(which('go'), bin_go):
+                # if Go is listed as a (build) dependency, use 'go' command provided that way
+                go = os.path.join(go_root, 'bin', 'go')
+                self.log.debug("Retaining 'go' command for Go dependency: %s", go)
+        if go:
+            self.go_cmd = go
+        else:
+            raise EasyBuildError("Failed to pick go command to use. Is it listed in dependencies?")
+
+        go_version = get_software_version('Go')
+        if LooseVersion(go_version) < LooseVersion("1.11"):
+            raise EasyBuildError("Go version < 1.11 doesn't support installing modules from go.mod")
+
+        # enforce use of go modules
+        env.setvar('GO111MODULE', 'on', verbose=False)
+        # set bin folder
+        env.setvar('GOBIN', os.path.join(self.installdir, 'bin'), verbose=False)
+
+    def configure_step(self):
+        """Configure Go package build/install."""
+
+        if self.go_cmd is None:
+            self.prepare_go()
+
+        # creates log entries for go being used, for debugging
+        run_cmd("%s version" % self.go_cmd, verbose=False, trace=False)
+        run_cmd("%s env" % self.go_cmd, verbose=False, trace=False)
+
+    def build_step(self):
+        """If Go package is not native go module, lets try to make the module."""
+
+        go_mod_file = os.path.join('./', 'go.mod')
+        go_sum_file = os.path.join('./', 'go.sum')
+
+        if not os.path.exists(go_mod_file) or not os.path.isfile(go_mod_file):
+            self.log.warn("go.mod not found! This is not natively supported go module. Trying to init module.")
+
+            if self.cfg['modulename'] is None:
+                raise EasyBuildError("Installing non-native go module. You need to specify 'modulename' in easyconfig")
+
+            # for more information about migrating to go modules
+            # see: https://blog.golang.org/migrating-to-go-modules
+
+            # go mod init
+            cmd = ' '.join([self.go_cmd, 'mod init', self.cfg['modulename']])
+            run_cmd(cmd, log_all=True, simple=True)
+
+            if self.cfg['forced_deps']:
+                for dep in self.cfg['forced_deps']:
+                    # go get specific dependencies which locks them in go.mod
+                    cmd = ' '.join([self.go_cmd, 'get %s@%s' % dep])
+                    run_cmd(cmd, log_all=True, simple=True)
+
+            # go build ./...
+            cmd = ' '.join([self.go_cmd, 'build ./...'])
```

Comment:
```python
cmd = ' '.join([self.go_cmd, 'build', './...'])
```

same below

(bonus points if you can add a comment to explain the triple dot, it looks like a typo...)